app/models/patient.rb

```diff
@@ -22,4 +23,37 @@ def has_entity_id?
   def self.entity_scope
     "patient"
   end
+
+  def self.attribute_field(*args)
+    args.each do |arg|
+      if self.entity_fields.detect { |f| f.name == arg.to_s }.pii?
+        define_method arg do
+          self.plain_sensitive_data[arg.to_s]
+        end
+
+        define_method "#{arg}=" do |value|
+          if value.blank?
+            self.plain_sensitive_data.delete(arg.to_s)
+          else
+            self.plain_sensitive_data[arg.to_s] = value
+          end
+        end
+      else
+        define_method arg do
+          self.core_fields[arg.to_s]
+        end
+
+        define_method "#{arg}=" do |value|
+          if value.blank?
+            self.core_fields.delete(arg.to_s)
+          else
+            self.core_fields[arg.to_s] = value
+          end
+        end
+      end
+    end
+  end
+
+  attribute_field :name, :gender, :dob, :email, :phone
+
```

Comment:
We can, yet is only used in Patient at the moment. Also we need to add a way to copy a field to an AR attribute.  Maybe a `copy: true` as an options. That will pollute a bit the above code most likely.